Docs: Restore "Migrate from Cortex" for Jsonnet (#3929)
grafana/mimir: docs/sources/mimir/migration-guide/migrate-from-cortex.md

---
title: "Migrate from Cortex to Grafana Mimir"
menuTitle: "Migrate from Cortex"
description: "Learn how to migrate your deployment of Cortex to Grafana Mimir to simplify the deployment and continued operation of a horizontally scalable, multi-tenant time series database with long-term storage."
weight: 10
---

# Migrate from Cortex to Grafana Mimir

This document guides an operator through the process of migrating a Jsonnet deployment of [Cortex](https://cortexmetrics.io/) to Grafana Mimir.
This document includes an overview of the steps required for any environment, and specific instructions for [environments deployed with Jsonnet](#migrating-to-grafana-mimir-using-jsonnet).
To migrate a Helm deployment of Cortex refer to [Migrate from Cortex](/docs/helm-charts/mimir-distributed/latest/migration-guides/migrate-from-cortex).

> Grafana Mimir 2.0.0 includes significant changes that simplify the deployment and continued operation of a horizontally scalable, multi-tenant time series database with long-term storage.

Reviewer: ```suggestion
Grafana Mimir includes significant changes that simplify the deployment and continued operation of a horizontally scalable, multi-tenant time series database with long-term storage.
```

- Removing configuration parameters that don't require tuning
- Renaming some parameters so that they're more easily understood
- Updating the default values of some existing parameters

The `mimirtool` automates configuration conversion.
It provides a simple migration by generating Mimir configuration from Cortex configuration.

## Before you begin

- Ensure that you are running either Cortex 1.10.X or Cortex 1.11.X.

  If you are running an older version of Cortex, upgrade to [1.11.1](https://github.com/cortexproject/cortex/releases) before proceeding with the migration.

- Ensure you have installed Cortex alerting and recording rules as well as Cortex dashboards.

  The monitoring mixin has both alerting and recording rules to install in either Prometheus or Cortex as well as dashboards to install in Grafana.
  To download a prebuilt ZIP file that contains the alerting and recording rules, refer to [Release Cortex-jsonnet 1.11.0](https://github.com/grafana/cortex-jsonnet/releases/download/1.11.0/cortex-mixin.zip).

  To upload rules to the ruler using mimirtool, refer to [mimirtool rules]({{< relref "../operators-guide/tools/mimirtool.md" >}}).

>   To import the dashboards into Grafana, refer to [Import dashboard](/docs/grafana/latest/dashboards/export-import/#import-dashboard).

Author: The doc validator is failing here, but this is the same string as used in the Helm docs, so I'm not sure what to do. Ideas, @osg-grafana ?

  - Alertmanager endpoints

    | Legacy                  | Current                            |
    | ----------------------- | ---------------------------------- |
    | `/<legacy-http-prefix>` | `/alertmanager`                    |
    | `/status`               | `/multitenant_alertmanager/status` |

## Generating configuration for Grafana Mimir

[`mimirtool`]({{< relref "../operators-guide/tools/mimirtool.md" >}}) provides a command for converting Cortex configuration to Mimir configuration that you can use to update both flags and configuration files.

### Install mimirtool

To install Mimirtool, download the appropriate binary from the [latest release](https://github.com/grafana/mimir/releases/latest) for your operating system and architecture and make it executable.

Alternatively, use a command line tool such as `curl` to download `mimirtool`. For example, for Linux with the AMD64 architecture, use the following command:

```bash
curl -fLo mimirtool https://github.com/grafana/mimir/releases/latest/download/mimirtool-linux-amd64
chmod +x mimirtool
```

### Use mimirtool

> The `mimirtool config convert` command converts Cortex 1.11 configuration files to Grafana Mimir 2.3 configuration files.

Reviewer: ```suggestion
The `mimirtool config convert` command converts Cortex 1.11 configuration files to Grafana Mimir configuration files.
```

---
title: "Migrate from Cortex to Grafana Mimir"
menuTitle: "Migrate from Cortex"
description: "Learn how to migrate your deployment of Cortex to Grafana Mimir to simplify the deployment and continued operation of a horizontally scalable, multi-tenant time series database with long-term storage."
weight: 10
---

# Migrate from Cortex to Grafana Mimir

> This document guides an operator through the process of migrating a Jsonnet deployment of [Cortex](https://cortexmetrics.io/) to Grafana Mimir.

Reviewer 2: ```suggestion
As an operator, you can migrate a Jsonnet deployment of [Cortex](https://cortexmetrics.io/) to Grafana Mimir.
```

> This document includes an overview of the steps required for any environment, and specific instructions for [environments deployed with Jsonnet](#migrating-to-grafana-mimir-using-jsonnet).

Reviewer 2: ```suggestion
The overview includes the steps required for any environment. To migrate deployment environments with Jsonnet, see [Migrate to Grafana Mimir using Jsonnent]({{< relref "#migrate-to-grafana-mimir-using-jsonnet" >}}).
```

> This document was tested with Cortex versions 1.10 and 1.11. While it may work with more recent versions of Cortex, we cannot guarantee that.
> To migrate a Helm deployment of Cortex refer to [Migrate from Cortex](/docs/helm-charts/mimir-distributed/latest/migration-guides/migrate-from-cortex).

Reviewer 2: ```suggestion
> **Note:** This document was tested with Cortex versions 1.10 and 1.11. It might work with more recent versions of Cortex, but that is not guaranteed.
To migrate a Helm deployment of Cortex refer to [Migrate from Cortex]({{< relref "/docs/helm-charts/mimir-distributed/latest/migration-guides/migrate-from-cortex" >}}).
```

Reviewer 2: Not sure this relref is going to work, but use relrefs whenever possible because they are testing during CI.

Author: the relref didn't work. I will revert the relref change

Reviewer 2: Yeah, I did not mean to change the order.

Author: ok, i will use the previous order

Grafana Mimir includes significant changes that simplify the deployment and continued operation of a horizontally scalable, multi-tenant time series database with long-term storage.

> The changes focus on making Grafana Mimir easier to run out of the box, including:

Reviewer 2: ```suggestion
The changes make Grafana Mimir easier to run out of the box:
```

> - Removing configuration parameters that don't require tuning
> - Renaming some parameters so that they're more easily understood
> - Updating the default values of some existing parameters

Reviewer 2: ```suggestion
- Removed configuration parameters that don't require tuning
- Renamed some parameters so that they're more easily understood
- Updated the default values of some existing parameters
```

The `mimirtool` automates configuration conversion.
It provides a simple migration by generating Mimir configuration from Cortex configuration.

> ## Before you begin

Reviewer 2: ```suggestion
**Before you begin:**
```
Style change since the previous version of this doc was written; see https://github.com/grafana/writers-toolkit/edit/main/docs/sources/writing-guide/topic-types/task/index.md.

- Ensure that you are running either Cortex 1.10.X or Cortex 1.11.X.

>   If you are running an older version of Cortex, upgrade to [1.11.1](https://github.com/cortexproject/cortex/releases) before proceeding with the migration.

Reviewer 2: ```suggestion
  If you are running an older version of Cortex, upgrade to [Cortex 1.11.1](https://github.com/cortexproject/cortex/releases) before proceeding with the migration.
```

- Ensure you have installed Cortex alerting and recording rules as well as Cortex dashboards.

>   The monitoring mixin has both alerting and recording rules to install in either Prometheus or Cortex as well as dashboards to install in Grafana.

Reviewer 2: ```suggestion
  For the monitoring mixin, you need to install both alerting and recording rules in either Prometheus or Cortex. You also need to install dashboards in Grafana.
```

Reviewer 2: Please vet this for technical accuracy. Does the monitoring mixin install alerting rules and recording rules, or does the use need to figure that out? Does my change make sense or have I misunderstood this?

If my change is technically inaccurate, let’s revisit the initial verbiage so we can make it clearer who or what is doing what.

Reviewer 2: Does this mean, "The monitoring mixin contains both alerting and recording rules that you {can | need to} install in either Prometheus or Cortex. The mixin also contains dashboards that you {can | need to} install in Grafana."

If I have a choice, why would I make one choice or the other wrt {can | need to}?

Reviewer 2: LGTM, thanks!

  To download a prebuilt ZIP file that contains the alerting and recording rules, refer to [Release Cortex-jsonnet 1.11.0](https://github.com/grafana/cortex-jsonnet/releases/download/1.11.0/cortex-mixin.zip).

  To upload rules to the ruler using mimirtool, refer to [mimirtool rules]({{< relref "../operators-guide/tools/mimirtool.md" >}}).
  To import the dashboards into Grafana, refer to [Import dashboard](/docs/grafana/latest/dashboards/export-import/#import-dashboard).

## Notable changes

> > **Note:** For full list of changes, refer to the project [CHANGELOG](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

Reviewer 2: ```suggestion
> **Note:** For the full list of changes, refer to Mimir’s [CHANGELOG](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
```

    | `/<legacy-http-prefix>/rules/{namespace}`             | `<prometheus-http-prefix>/config/v1/rules/{namespace}`             |
    | `/<legacy-http-prefix>/rules/{namespace}/{groupName}` | `<prometheus-http-prefix>/config/v1/rules/{namespace}/{groupName}` |
    | `/<legacy-http-prefix>/rules/{namespace}`             | `<prometheus-http-prefix>/config/v1/rules/{namespace}`             |
    | `/<legacy-http-prefix>/rules/{namespace}/{groupName}` | `<prometheus-http-prefix>/config/v1/rules/{namespace}/{groupName}` |
    | `/<legacy-http-prefix>/rules/{namespace}`             | `<prometheus-http-prefix>/config/v1/rules/{namespace}`             |
    | `/ruler_ring`                                         | `/ruler/ring`                                                      |

  - Alertmanager endpoints

    | Legacy                  | Current                            |
    | ----------------------- | ---------------------------------- |
    | `/<legacy-http-prefix>` | `/alertmanager`                    |
    | `/status`               | `/multitenant_alertmanager/status` |

> ## Generating configuration for Grafana Mimir

Reviewer 2: ```suggestion
## Generate the configuration for Grafana Mimir
```
Default to gerunds.

> [`mimirtool`]({{< relref "../operators-guide/tools/mimirtool.md" >}}) provides a command for converting Cortex configuration to Mimir configuration that you can use to update both flags and configuration files.

Reviewer 2: ```suggestion
The [`mimirtool`]({{< relref "../operators-guide/tools/mimirtool.md" >}}) command converts Cortex configuration to Mimir configuration that you can use to update both flags and configuration files.
```

Author: how about this?

```suggestion
The [`mimirtool`]({{< relref "../operators-guide/tools/mimirtool.md#config" >}}) `config` command converts Cortex configuration to Mimir configuration. You can use it to update both flags and configuration files.
```

Reviewer 2: Based on https://github.com/grafana/mimir/pull/3929/files#diff-7c7a10d55fd0934f69c75dee10a5c77ebeb80d0764d92529882c930664892560R122, the command looks to be `mimirtool config convert`. In that case, I recommend:

The [`mimirtool config convert`]({{< relref "../operators-guide/tools/mimirtool.md#config" >}}) command converts Cortex configuration to Mimir configuration. You can use it to update both flags and configuration files.

Alternatively, use a command line tool such as `curl` to download `mimirtool`. For example, for Linux with the AMD64 architecture, use the following command:

```bash
curl -fLo mimirtool https://github.com/grafana/mimir/releases/latest/download/mimirtool-linux-amd64
chmod +x mimirtool
```

### Use mimirtool

The `mimirtool config convert` command converts Cortex 1.11 configuration files to Grafana Mimir configuration files.
It removes any configuration parameters that are no longer available in Grafana Mimir, and it renames configuration parameters that have a new name.
If you have explicitly set configuration parameters to a value matching the Cortex default, by default, `mimirtool config convert` doesn't update the value.
To have `mimirtool config convert` update explicitly set values from the Cortex defaults to the new Grafana Mimir defaults, provide the `--update-defaults` flag.
Refer to [convert]({{< relref "../operators-guide/tools/mimirtool.md#convert" >}}) for more information on using `mimirtool` for configuration conversion.

> ## Migrating to Grafana Mimir using Jsonnet

Reviewer 2: ```suggestion
## Migrate to Grafana Mimir using Jsonnet
```
